src/matrices.jl

```diff
@@ -351,23 +165,57 @@ function JointRecurrenceMatrix(x, y, ε; kwargs...)
     return JointRecurrenceMatrix(rm1.data .* rm2.data)
 end
 
+"""
+    JointRecurrenceMatrix(R1, R2; kwargs...)
+Create a joint recurrence matrix from given recurrence matrices `R1, R2`.
+"""
+function JointRecurrenceMatrix(R1::AbstractRecurrenceMatrix, R2::AbstractRecurrenceMatrix; kwargs...)
+    R3 = R1.data .* R2.data
+    return JointRecurrenceMatrix(R3)
+end
 
 ################################################################################
 # Scaling / fixed rate
 ################################################################################
 # here args... is (x, y, metric, ε) or just (x, metric, ε)
-function resolve_scale(args...; scale=1, fixedrate=false)
+function resolve_scale(args...; scale=1, fixedrate=false, fan=false)
     ε = args[end]
+    x = args[1]
```

Comment:
I'd suggest that instead of defining the arguments for `get_fan_threshold` based on the length of `args`, the function `get_fan_threshold` could be redefined with different methods (for three or four arguments), such that `args` can be passed directly to it.

Reply:
yepp, also right. I couldn't simply help myself in the first turn, because I did not (and I think still do not) know much about string interning.